### PR TITLE
[Update]: Add LinkedIn as a provider

### PR DESCRIPTION
**Pull request by mikah13**

### CODE REVIEW

**Comment by pilcrowonpaper** [comment]
I recently updated the source code to include `class GitHub implements OAuth2Provider` etc. Also, can you update the README?

**Comment by mikah13** [comment]
Hi, i have added the Provider for the LinkedIn class. I notice there is a section for Others that got repeated twice in the README so I just removed it too. Please review, ty !

File: README.md [hunk 1]
```diff
@@ -98,13 +99,6 @@ try {
 }
 ```
 
-### Other
```
Comment (pilcrowonpaper) [inline README.md]: Why was this removed?

File: src/providers/linkedin.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		clientSecret: string,
+		redirectURI: string,
+		options?: {
+			scope?: string[];
+		}
+	) {
+		this.client = new OAuth2Client(clientId, authorizeEndpoint, tokenEndpoint, {
+			redirectURI
+		});
+		this.scope = options?.scope ?? [];
```
Comment (pilcrowonpaper) [inline src/providers/linkedin.ts]:
```suggestion
		this.scope = options?.scope ?? [];
		this.scope.push("openid", "profile");
```

File: src/providers/linkedin.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	refresh_token_expires_in: number;
+	scope: string;
+}
+
+export type LinkedInTokens = {
+	accessToken: string;
+	accessTokenExpiresIn: number;
+	refreshToken: string;
+	refreshTokenExpiresIn: number;
+	scope: string;
```
Comment (pilcrowonpaper) [inline src/providers/linkedin.ts]: I don't think `scope` is really necessary
```suggestion
```

**Comment by pilcrowonpaper** [comment]
Thanks!
